components/Admin/Admin.js

```diff
@@ -0,0 +1,65 @@
+import React, { PropTypes } from 'react'
+import AdminTopBar from '../AdminTopBar'
+import LocationList from '../LocationList'
+import history from '../../core/history'
+import { fetchLocations } from '../../core/firebaseApi'
+
+class Admin extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      locations: [],
+      filteredLocations: null,
+    }
+  }
+
+  componentWillMount() {
+    fetchLocations()
+      .then(locations => {
+        this.setState({ locations })
+      })
+  }
+
+  handleSearch = (event) => {
+    const searchTerm = event.currentTarget.value
+    let { locations, filteredLocations } = this.state
+
+    if (searchTerm) {
+      filteredLocations = locations.filter(location =>
+        location.name.indexOf(searchTerm) >= 0)
+    } else {
+      filteredLocations = null
+    }
+
+    this.setState({
+      locations,
+      filteredLocations,
+    })
+  }
+
+  handleNewFacility = () => {
```

Comment:
For others who are reviewing, this is a way of "autobinding" methods in JS. It's very much experimental ESNext. Another solution is in the constructor to do:
```javascript
this.handleSearch = this.handleSearch.bind(this)
```

There are many ways to do this and it's annoying but necessary to have anything work.
https://www.sitepoint.com/bind-javascripts-this-keyword-react/